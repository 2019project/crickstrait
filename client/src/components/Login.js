import React, { Component } from "react";
import { login } from "../actions/Users";
import "../styles/Login.css";
import { Link } from "react-router-dom";

import { connect } from "react-redux";

export class Login extends Component {
  state = {
    email: "",
    password: "",
    errors: {},
    login: true,
    signup: true
  };

  OnChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  // Login Function to be called on Click
  onLogin = e => {
    e.preventDefault();
    // Checks the validation
    if (this.validateForm()) {
      let user = {
        email: this.state.email,
        password: this.state.password
      };
      this.props.login(user, this.props.history);
      this.setState({
        email: "",
        password: ""
      });
    }
  };

  // Function for validation check
  validateForm() {
    console.log("Hello");
    let errors = {};
    let formIsValid = true;
    if (!this.state.email) {
      formIsValid = false;
      errors["email"] = "*Please enter your Email-ID.";
    }

    if (!this.state.password) {
      formIsValid = false;
      errors["password"] = "*Please enter your Password.";
    }

    this.setState({
      errors: errors
    });
    return formIsValid;
  }

  render() {
    return (
      <div>
        {/* Main Container For login */}
        <div id="loginMainContainer">
          {/* Login Left Container for Welcome Text */}
          <div id="loginLeftContainer">
            <h1
              id="loginwelcome"
              className="w3-container w3-center w3-animate-left"
            >
              Welcome ,
            </h1>
            <p className="w3-container w3-center w3-animate-left">
              An Investment in Knowledge <br />
              pays the best Interest
            </p>
          </div>

          {/* Login Component Right container login form */}
          <div id="loginRightContainer">
            {/* Container for Buttons */}
            <div>
              <Link to="/login">
                <button
                  id={
                    this.state.login
                      ? "loginComponentButton-active"
                      : "loginComponentButton"
                  }
                  onClick={() => {
                    this.setState({
                      login: false,
                      signup: true
                    });
                  }}
                >
                  Login
                </button>
              </Link>
              <Link to="/register">
                <button id="registerComponentButton">Sign Up</button>
              </Link>
            </div>

            <form id="loginForm">
              <p>
                {" "}
                <input
                  id="loginInput"
                  placeholder="Email"
                  type="text"
                  name="email"
                  onChange={this.OnChange}
                  value={this.state.email}
                />
              </p>
              <p id="loginErrorMessage">{this.state.errors.email}</p>
              <p>
                <input
                  id="loginInput"
                  placeholder="Password"
                  type="password"
                  name="password"
                  onChange={this.OnChange}
                  value={this.state.password}
                />
              </p>
              <p id="loginErrorMessage">{this.state.errors.password}</p>

              {this.props.error ? (
                <p id="loginErrorMessage">{this.props.error}</p>
              ) : (
                <p style={{ display: "none" }}>Hello</p>
              )}
              <Link to="/resetpassword">
                <p
                  style={{
                    color: "#39abf7"
                  }}
                >
                  Forgot Password ?
                </p>
              </Link>
              <p>
                <button onClick={this.onLogin} id="loginFormButton">
                  Login
                </button>
              </p>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  users: state.usersReducer.users,
  error: state.usersReducer.error
});

export default connect(
  mapStateToProps,
  { login }
)(Login);
