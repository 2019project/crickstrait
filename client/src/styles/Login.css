@import url(https://fonts.googleapis.com/css?family=Karla);
@import url("https://www.w3schools.com/w3css/4/w3.css");
@import url("https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css");
* {
  font-family: Karla;
}

#loginMainContainer {
  width: 100%;
  margin: 0px auto;
  height: 510px;
  grid-template-columns: "side1";
  background-color: #3e5871;
  display: grid;
  text-align: center;
}

#loginLeftContainer {
  color: #fff;
  font-size: 18px;
}

#welcome {
  font-size: 34px;
}

#loginRightContainer {
  color: #fff;
  padding: 5 px;
}

#loginInput {
  border: none;
  outline: none;
  background: transparent;
  color: #fff;
  border-bottom: 1px solid #cacaca;
  padding: 15px;
  width: 70%;
  font-size: 16px;
}

#loginInput::placeholder {
  color: #b0b0b0;
}

#loginInput:focus,
#loginInput:active {
  transform: scaleX(1.1);
  border-bottom: 2px solid #39abf7;
}

#loginFormButton {
  padding: 15px;
  background: transparent;
  outline: none;
  font-size: 16px;
  border: 3px solid #39abf7;
  color: #fff;
  margin-top: 20px;

  -webkit-transition-duration: 0.4s; /* Safari */
  transition-duration: 0.4s;
  cursor: pointer;
}

#loginFormButton:hover {
  right: 0;
  background-color: #39abf7;
  box-shadow: 0 12px 16px 0 rgba(231, 226, 226, 0.24),
    0 17px 50px 0 rgba(231, 226, 226, 0.24);
}

#loginFormButton::after {
  content: "";
  display: inline-block;
  height: 100%;
  width: 100%;
  border-radius: 100px;
  position: absolute;
  top: 0;
  left: 0;
  z-index: -1;
  transition: all 0.4s;
}

#loginErrorMessage {
  color: #ff4d4d;
  font-size: 14px;
}

#loginComponentButton {
  background: transparent;
  color: #cacaca;
  outline: none;
  padding: 20px;
  cursor: pointer;
  font-weight: 500;
  font-size: 20px;
  border: none;
}

#loginComponentButton-active {
  background: transparent;
  color: #fff;
  outline: none;
  padding: 20px;
  font-size: 20px;
  border: none;
  border-bottom: 2px solid #39abf7;
}

#registerComponentButton-active {
  background: transparent;
  color: #fff;
  outline: none;
  padding: 20px;
  font-size: 20px;
  border: none;
  border-bottom: 2px solid #39abf7;
}

#registerComponentButton {
  background: transparent;
  color: #cacaca;
  cursor: pointer;
  outline: none;
  padding: 20px;
  font-weight: 500;
  font-size: 20px;

  border: none;
}

#registerResetPassword {
  margin-left: 310px;
  color: #39abf7;
}

@media screen and (min-width: 430px) {
  #loginMainContainer {
    height: 675px;
    grid-template-columns: repeat(2, 1fr);
  }
  #loginMainContainer > div:nth-child(2) > form input {
    width: 50%;
  }

  #loginLeftContainer {
    grid-column: 1/2;
  }

  #loginRightContainer {
    grid-column: 2/-1;
  }

  #loginMainContainer > div:nth-child(2) {
    margin: auto 0px;
  }

  #loginMainContainer > div:nth-child(1) {
    margin: 150px auto;
  }

  #loginLeftContainer > h1 {
    font-size: 60px;
  }

  #loginLeftContainer > p {
    font-size: 40px;
  }
}
