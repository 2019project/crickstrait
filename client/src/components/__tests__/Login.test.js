import React from "react";
import { shallow, mount } from "enzyme";
import { Login } from "../Login";
const login = jest.fn();

const wrapper = shallow(<Login login={login} />);

describe("Testing Login Component", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it("should mount the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have a container with name loginMainContainer", () => {
    expect(wrapper.find("#loginMainContainer")).toBeTruthy();
  });

  it("should have a container with name loginLeftContainer", () => {
    expect(wrapper.find("#loginLeftContainer")).toBeTruthy();
  });

  it("should have a container with name loginRightContainer", () => {
    expect(wrapper.find("#loginRightContainer")).toBeTruthy();
  });

  it("should simulate button click on Login button tab ", () => {
    wrapper.find("#loginComponentButton-active").simulate("click");
    expect(wrapper.state().login).toBe(false);
    expect(wrapper.state().signup).toBe(true);
  });

  it("checks for on edit button function to be called", () => {
    const e = { preventDefault: () => {} };
    jest.spyOn(e, "preventDefault");
    wrapper.find("#loginFormButton").simulate("click", e);
    expect(e.preventDefault).toBeCalled();
  });

  it("should check for presense of onLogin function", () => {
    let onLogin = jest.spyOn(wrapper.instance(), "onLogin");
    expect(onLogin).toBeTruthy();
  });

  it("should check for presense of OnChange function", () => {
    let OnChange = jest.spyOn(wrapper.instance(), "OnChange");
    expect(OnChange).toBeTruthy();
  });
});
