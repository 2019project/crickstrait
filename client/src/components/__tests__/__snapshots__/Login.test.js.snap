// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Login Component should mount the component 1`] = `
<div>
  <div
    id="loginMainContainer"
  >
    <div
      id="loginLeftContainer"
    >
      <h1
        className="w3-container w3-center w3-animate-left"
        id="loginwelcome"
      >
        Welcome ,
      </h1>
      <p
        className="w3-container w3-center w3-animate-left"
      >
        An Investment in Knowledge 
        <br />
        pays the best Interest
      </p>
    </div>
    <div
      id="loginRightContainer"
    >
      <div>
        <Link
          to="/login"
        >
          <button
            id="loginComponentButton-active"
            onClick={[Function]}
          >
            Login
          </button>
        </Link>
        <Link
          to="/register"
        >
          <button
            id="registerComponentButton"
          >
            Sign Up
          </button>
        </Link>
      </div>
      <form
        id="loginForm"
      >
        <p>
           
          <input
            id="loginInput"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="text"
            value=""
          />
        </p>
        <p
          id="loginErrorMessage"
        />
        <p>
          <input
            id="loginInput"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
        </p>
        <p
          id="loginErrorMessage"
        />
        <p
          style={
            Object {
              "display": "none",
            }
          }
        >
          Hello
        </p>
        <Link
          to="/resetpassword"
        >
          <p
            style={
              Object {
                "color": "#39abf7",
                "marginTop": "20px",
              }
            }
          >
            Forgot Password ?
          </p>
        </Link>
        <p>
          <button
            id="loginFormButton"
            onClick={[Function]}
          >
            Login
          </button>
        </p>
      </form>
    </div>
  </div>
</div>
`;
